components/ruby/lib/chef-licensing/restful_client/base.rb

```diff
@@ -176,6 +199,46 @@ def raise_restful_client_conn_error(urls)
 
         raise RestfulClientConnectionError, error_message
       end
+
+      def construct_cache_key(endpoint, params = {})
+        string_to_hash = "#{ChefLicensing::Config.license_server_url}_#{endpoint}"
```

Comment:
I don't understand this bit, if `ChefLicensing::Config.license_server_url` is an string of multiple URLs, how is this valid?

Reply:
`ChefLicensing::Config.license_server_url` would be a string of multiple URLs separated by commas only for the first run. 

However, this question made me think of a situation where the cache won't be fetched if the user has multiple license server url set in the system. Let me do some testing around this.

Reply:
We may need to remove the dependency on the license_server_url for the cache key construction.